bookkeeper-server/src/main/java/org/apache/bookkeeper/client/api/LedgerEntries.java

```diff
@@ -0,0 +1,52 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.bookkeeper.client.api;
+
+import java.util.Iterator;
+
+/**
+ * Interface to wrap the entries.
+ *
+ * @since 4.6
+ */
+public interface LedgerEntries extends AutoCloseable {
+
+    /**
+     * Gets a specific LedgerEntry by entryId.
+     *
+     * @param entryId the LedgerEntry id
+     * @return the LedgerEntry, null if no LedgerEntry with such entryId.
+     */
+    LedgerEntry getEntry(long entryId);
+
+    /**
+     * In this, It does not retain the ByteBuf references for the entries in this LedgerEntries.
+     * The caller who calls #iterator() should be careful for not releasing the references.
+     *
+     * @return the iterator of type LedgerEntry
+     */
+    Iterator<LedgerEntry> iterator();
+
+    /**
+     * In this, It retains the ByteBuf references for the entries in this LedgerEntries.
+     * The caller who calls #iterator() is responsible for releasing the retained references.
```

Comment:
change `#iterator()` to `{@link #retainIterator()}`

Reply:
Thanks. will change it.